ska_helpers/tests/test_git_helpers.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+import tempfile
+from ska_helpers import git_helpers
+import git
+import os
+from pathlib import Path
+
+
+CHANDRA_MODELS = Path(os.environ["SKA"]) / "data" / "chandra_models"
+
+
+def ska_ownership_ok():
+    # Check that the OS and volume support file ownership
+    # (not the case with shared directories on a Windows VM on parallels)
+    # and that the chandra_models dir is owned by the current user
+    try:
+        return CHANDRA_MODELS.owner() == os.getlogin()
+    except Exception:
+        return False
+
+
+@pytest.mark.skipif(
+    not CHANDRA_MODELS.exists(), reason="Chandra models dir is not there"
+)
+@pytest.mark.skipif(ska_ownership_ok(), reason="Chandra models dir ownership is OK")
+def test_make_git_repo_safe(monkeypatch):
+    with tempfile.TemporaryDirectory() as tempdir:
+        # temporarily set HOME to a temp dir so .gitconfig comes from there
+        monkeypatch.setenv("HOME", tempdir)
+        repo = git.Repo(CHANDRA_MODELS)
+        with pytest.raises(git.exc.GitCommandError):
+            # This statement fails with the error
+            #     fatal: detected dubious ownership
+            # unless the repo is marked safe in .gitconfig
+            repo.is_dirty()
+        with pytest.warns(UserWarning, match="Updating git config"):
+            # marke the repo safe and issue warning
+            git_helpers.make_git_repo_safe(CHANDRA_MODELS)
+        # success
```

Comment:
If this is using a temporary gitconfig to confirm that the make_git_repo_safe makes the repo safe via editing the temporary gitconfig, it seems like it would make sense to just read the gitconfig and confirm the right change was made.

Reply:
That's fine.  I don't disagree that checking that the repo is usable is also a good test and one that you would want here, I was just thinking for functional testing you want to see that what you asked for is what happened.  But if you are happy skipping to the end that's fine.